File: core/src/main/java/dungeon/crawler/GameSystem/SaveGame/Serialization/GameSave.java
package dungeon.crawler.GameSystem.SaveGame.Serialization;

import java.util.ArrayList;

public class GameSave {
    public PartyCharacterSave player;
    public int overWorldCoordinatesX;
    public int overWorldCoordinatesY;
    public ArrayList<PartyCharacterSave> party;

    public int gold;
    public boolean isPlayerDead;
    public ArrayList<String> bag;
    public GameSave(){
    }
}

File: core/src/main/kotlin/dungeon/crawler/GameSystem/SaveGame/Serialization/GameSave.kt
package dungeon.crawler.GameSystem.SaveGame.Serialization

import dungeon.crawler.GameSystem.GameBuild
import kotlinx.serialization.Serializable

@Serializable
data class GameSave(
    val player: PartyCharacterSave,
    val isPlayerDead: Boolean,
    val currentMap: String,
    val ScreenID: Int,
    val Bag: ArrayList<String>,
    val overWorldCoordinatesX: Int,
    val overWorldCoordinatesY: Int,
    val party: ArrayList<PartyCharacterSave>,
    val gold: Int
)
